keepassxc-browser/common/sites.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+// List of sites that need special handling. 'rep' is the replacement URL.
+const siteList = [
+    {
+        url: 'accounts.google.com',
+        rep: 'https://accounts.google.com/*'
+    },
+    {
+        url: 'www.paypal.com',
+        rep: 'https://www.paypal.com/*'
+    },
+    {
+        url: 'login.live.com',
+        rep: 'https://login.live.com/*'
+    },
+    {
+        url: 'amazon.com',
+        rep: 'https://www.amazon.com/*'
+    },
+    {
+        url: 'amazon.de',
+        rep: 'https://www.amazon.de/*'
+    },
+    {
+        url: 'amazon.co.uk',
+        rep: 'https://www.amazon.co.uk/*'
+    },
+    {
+        url: 'signin.aws.amazon.com',
+        rep: 'https://signin.aws.amazon.com/*'
+    },
+    {
+        url: 'www.upwork.com/ab/',
+        rep: 'https://www.upwork.com/ab/*'
+    },
+    {
+        url: 'home.personalcapital.com',
+        rep: 'https://home.personalcapital.com/*'
+    }
+];
+
+const kpxcSites = {};
+
+// Returns a predefined URL for certain sites to ensure compatibility with Site Preferences
+kpxcSites.definedURL = function(url) {
+    for (const site of siteList) {
+        if (url.includes(site.url)) {
+            return site.rep;
+        }
+    }
+
+    return url;
+};
+
+// Adds all common sites with multi-page login to Site Preferences
+kpxcSites.addAllCommonSites = function(settings) {
+    for (const site of siteList) {
+        if (settings['sitePreferences'].some(e => e.url === site.rep)) {
+            continue;
+        }
+
+        settings['sitePreferences'].push({
```

Comment:
I don't think we should actually be pushing the site list to the site preference list. That list is user domain and we are trashing their settings. We should check both the site preferences list and the "default list" (which is just an internal array) if its enabled.

Reply:
Good idea!

Reply:
And also based from the conversation at IRC: a separate plain text file for the list, if possible.